renegade/client.py

```diff
@@ -0,0 +1,125 @@
+        return cls(api_key, api_secret, SEPOLIA_BASE_URL)
+
+    @classmethod
+    def new_mainnet_client(cls, api_key: str, api_secret: str) -> "ExternalMatchClient":
+        return cls(api_key, api_secret, MAINNET_BASE_URL)
+
+    async def request_quote(self, order: ExternalOrder) -> Optional[SignedExternalQuote]:
+        request = ExternalQuoteRequest(external_order=order)
+
+        headers = self._get_headers()
+        response = await self.http_client.post_with_headers(REQUEST_EXTERNAL_QUOTE_ROUTE, request.model_dump(), headers)
+        quote_resp = await self._handle_optional_response(response)
+        if quote_resp:
+            return ExternalQuoteResponse(**quote_resp).signed_quote
+
+        return None
+
+    async def assemble_quote(self, quote: SignedExternalQuote) -> Optional[AtomicMatchApiBundle]:
+        return await self.assemble_quote_with_options(quote, ExternalMatchOptions())
+
+    async def assemble_quote_with_options(
+        self, 
+        quote: SignedExternalQuote, 
+        options: ExternalMatchOptions
+    ) -> Optional[AtomicMatchApiBundle]:
+        request = AssembleExternalMatchRequest(
+            do_gas_estimation=options.do_gas_estimation,
+            receiver_address=options.receiver_address,
+            signed_quote=quote
+        )
+
+        headers = self._get_headers()
+        response = await self.http_client.post_with_headers(ASSEMBLE_EXTERNAL_MATCH_ROUTE, request.model_dump(), headers)
+        match_resp = await self._handle_optional_response(response)
+        if match_resp:
+            return ExternalMatchResponse(**match_resp).match_bundle
+
+        return None
+
+    async def request_external_match(self, order: ExternalOrder) -> Optional[AtomicMatchApiBundle]:
+        return await self.request_external_match_with_options(order, ExternalMatchOptions())
+
+    async def request_external_match_with_options(
+        self, 
+        order: ExternalOrder, 
+        options: ExternalMatchOptions
+    ) -> Optional[AtomicMatchApiBundle]:
+        request = ExternalMatchRequest(
+            do_gas_estimation=options.do_gas_estimation,
+            receiver_address=options.receiver_address,
+            external_order=order
+        )
+        headers = self._get_headers()
+        response = await self.http_client.post_with_headers(REQUEST_EXTERNAL_MATCH_ROUTE, request.model_dump(), headers)
+        match_resp = await self._handle_optional_response(response)
+        if match_resp:
+            return ExternalMatchResponse(**match_resp).match_bundle
+
+        return None
+
+    def _get_headers(self) -> Headers:
+        headers = Headers()
+        headers[RENEGADE_API_KEY_HEADER] = self.api_key
+        return headers
+
+    async def _handle_optional_response(self, response: Response) -> Optional[dict]:
+        if response.status_code == 204:  # NO_CONTENT
+            return None
+        elif response.status_code == 200:  # OK
+            return response.json()
```

Comment:
not too familiar with `httpx` and how it does json parsing under the hood, but a lesson learned from the JS SDK: we should ensure that floating-point values in json responses are parsed with full precision

Reply:
we very intentionally do not send any floating points in the external match API, or we stringify them first to avoid precision loss. this is necessary because the quote is signed and so precision loss would destroy the signed payload